content/news/XXXX-XX-XX-jack-zero-latency.md

```diff
@@ -0,0 +1,74 @@
+
+Here the resulting recordings visualized in [Audacity](https://www.audacityteam.org):
+
+![Screenshot of audacity showing the round trip latency]({static}/images/news/roundtriplatency.png)
+
+The upper stream is the JACK case. The left channel is the recorded master 440 Hz sine wave and the right channel is the mic input.
+
+JACK is configured with a 1024 frames buffer and reports a latency of 46.4 ms for the sum of two buffers.
+The round trip latency is 95 ms ~4 buffers = Driver + ALSA + JACK Async mode + duplex stream
+
+The lower stream is the ALSA case. Mixxx is configures with the same single buffer of 1024 frames = 23.2 ms
+The round trip latency is 49 ms ~2 buffers = Driver + ALSA
+
+Not in the picture is the ALSA pulse device. It runs at a latency of 104 ms ~5 buffers. Driver + ALSA + 2 x Pulse + ALSA
+PipeWire on Fedora 34 has by default the same latency as JACK in this test.
+
+With this picture we can verify that Mixxx actually has the same buffer size in both cases. When pressing pause, it fades the signal out over one buffer length which is equal in both cases.
+The peaks in the recorded right channel is the sound of the mouse click. You can only barely see the recorded sine wave.
+
+![Screenshot of audacity showing the fade out]({static}/images/news/fadeoutcompare.png)
+
+For reference, I have done the same test using [jack_iodelay](http://manpages.ubuntu.com/manpages/bionic/man1/jack_iodelay.1.html)
+
+```
+  3114.842 frames     70.631 ms total roundtrip latency
+	extra loopback latency: 42 frames
+	use 21 for the backend arguments -I and -O Inv
+```
+
+The result is 70 ms ~3 buffers = Driver + ALSA + JACK Async mode. The duplex cycle is omitted here. This is one buffer more than a native ALSA implementation.
+
+To verify that the duplex cycle is not introduced in the Mixxx or [PortAudio](http://www.portaudio.com) abstraction layer used by Mixxx. I have routed the jack_iodelay signal trough Mixxx.
+
+![Jack patch field showing the jack_iodelay Mixxx loop]({static}/images/news/jackpatch.png)
+
+
+```
+  2048.000 frames     46.440 ms total roundtrip latency
+	extra loopback latency: 2047 frames
+	use 1023 for the backend arguments -I and -O
+```
+
+This is the minimum we can expect, one buffer is needed for jack_iodelay and one for Mixxx. The result can be confirmed with the same setup using jack_latent_client which just passes the input to the output.
+
+If we connect the output of Mixxx to the input, an extra buffer is used for some reasons. This can be also confirmed with jack_latent_client and the same feedback loop.
```

Comment:
```suggestion
If we connect the output of Mixxx to the input, an extra buffer is used for some reason. This can be also confirmed with `jack_latent_client` and the same feedback loop.
```